src/include/migraphx/instruction_ref.hpp

```diff
@@ -32,7 +32,40 @@ namespace migraphx {
 inline namespace MIGRAPHX_INLINE_NS {
 
 struct instruction;
+#if defined(_WIN32) && !defined(NDEBUG)
+class instruction_ref : public std::list<instruction>::iterator
+{
+public:
+    instruction_ref() = default;
+    instruction_ref(const std::list<instruction>::iterator& other)
+    {
+        _Ptr = other._Ptr;
+        _Adopt(other._Getcont());
+    }
+
+    bool operator==(const instruction_ref& other) const
+    {
+        return this->_Unwrapped()._Ptr == other._Unwrapped()._Ptr;
+    }
+
+    bool operator==(const std::list<instruction>::iterator& other) const
+    {
+        return this->_Unwrapped()._Ptr == other._Unwrapped()._Ptr;
+    }
+
+    bool operator==(const std::list<instruction>::const_iterator& other) const
+    {
+        return this->_Unwrapped()._Ptr == other._Unwrapped()._Ptr;
+    }
+
+    bool operator!=(const instruction_ref& other) const
```

Comment:
This also should be declared as `friend bool operator!=(const std::list<instruction>::iterator& x, const std::list<instruction>::iterator& y)`.